students/lauraannf/lessons/lesson04/assignment/basic_operations.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Apr  1 16:47:37 2019
+
+@author: Laura.Fiorentino
+"""
+
+import logging
+from customer_model import Customer
+
+LOG_FORMAT = "%(asctime)s %(filename)s:%(lineno)-3d %(levelname)s %(message)s"
+FORMATTER = logging.Formatter(LOG_FORMAT)
+FILE_HANDLER = logging.FileHandler('db.log')
+FILE_HANDLER.setFormatter(FORMATTER)
+LOGGER = logging.getLogger()
+LOGGER.addHandler(FILE_HANDLER)
+LOGGER.setLevel(logging.INFO)
+
+
+def add_customer(customer_id, first_name, last_name, home_address,
+                 phone_number, email_address, status, credit_limit):
+    """ adds customer to database"""
+    print('Adding new customer, Customer ID {}...'.format(customer_id))
+    try:
+        Customer.get_by_id(customer_id)
+        print('Customer ID {} is already in use'.format(customer_id))
+    except Exception:
```

Comment:
Avoid using "catch all" exception handling. The problem is that it will intercept the exceptions you can think of and the ones you can't, so you code could end up catching (and incorrectly handling) an unexpected exception. Please make sure you refactor all instances of catch all exception handling with specific exceptions.